src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs

```diff
@@ -411,6 +404,37 @@ public void RenameParameter(string expressionBase, string expectedExpression, st
             }
         }
 
+        [Theory]
+        [InlineData("firstos.option_1 <> Os1Value", "newOsName.Option1 <> Os1Value", "firstos", "newOsName")]        
+        [InlineData("firstos.option_1 <> firstos.option_2", "firstos.option_5 <> firstos.option_2", "firstos.option_1", "option_5")]
+        [InlineData("firstos.option_1 <> Os1Value", "firstos.option_1 <> Os1Value", "Unknown", "Unknown2")]
+        [InlineData("firstos.option_1 <> firstos.option_2", "firstos.option_1 <> firstos.option_2", "firstos.Unknown", "Unknown2")]
+        public void RenameParameter_OptionSet(string expressionBase, string expectedExpression, string oldName, string newName)
+        {
+            PowerFxConfig config = new PowerFxConfig();
+            OptionSet optionSet1 = new OptionSet("firstos", DisplayNameUtility.MakeUnique(new Dictionary<string, string>()
+            {
+                    { "option_1", "Option1" },
+                    { "option_2", "Option2" }
+            }));
+
+            config.AddOptionSet(optionSet1, new DName("TestFirstOptionSet"));
+
+            RecordType r1 = RecordType.Empty()                
+                .Add(new NamedFormulaType("Os1Value", optionSet1.FormulaType, "DisplayOS1Value"));
+
+            DPath oldNameAsPath = DPath.Root;
+            foreach (var segment in oldName.Split('.'))
+            {
+                oldNameAsPath = oldNameAsPath.Append(new DName(segment));
+            }
+
+            Engine engine = new Engine(config);
+
+            RenameDriver renamer = engine.CreateFieldRenamer(r1, oldNameAsPath, new DName(newName), CultureInfo.InvariantCulture);
```

Comment:
Interesting calling pattern. What if there's a field matching oldName as well as an optionSet? A method on engine: `createOptionSetRenamer` may be a better approach.